Makefile

```diff
@@ -66,46 +59,12 @@ gofuzz_json:
 fmt:
 	@go fmt ./...
 
-metalinter:
-	@echo "==> Running linter"
-	gometalinter.v2 --vendor --deadline=600s --disable-all  \
-		--enable=deadcode \
-		--enable=goconst \
-		--enable=goimports \
-		--enable=gosimple \
-		--enable=ineffassign \
-		--enable=megacheck \
-		--enable=misspell \
-		--enable=staticcheck \
-		--enable=safesql \
-		--enable=structcheck \
-		--enable=unconvert \
-		--enable=unused \
-		--enable=varcheck \
-		--enable=vetshadow \
-		./...
-
-		#--enable=maligned \
-		#--enable=gas \
-		#--enable=aligncheck \
-		#--enable=dupl \
-		#--enable=errcheck \
-		#--enable=gocyclo \
-		#--enable=golint \ <== comments on anything exported
-		#--enable=gotype \
-		#--enable=interfacer \
-		#--enable=unparam \
-		#--enable=vet \
-
-metalinter_all:
-	protoc $(INCLUDE) --lint_out=. types/*.proto
-	gometalinter.v2 --vendor --deadline=600s --enable-all --disable=lll ./...
-
-
-test_golang1.10rc:
-	docker run -it -v "$(CURDIR):/go/src/github.com/tendermint/go-amino" -w "/go/src/github.com/tendermint/go-amino" golang:1.10-rc /bin/bash -ci "make get_tools all"
+# look into .golangci.yml for enabling / disabling linters
+lint:
+	@echo "--> Running linter"
+	@golangci-lint run
 
 # To avoid unintended conflicts with file names, always add to .PHONY
 # unless there is a reason not to.
 # https://www.gnu.org/software/make/manual/html_node/Phony-Targets.html
-.PHONY: build install check_tools get_tools update_tools get_vendor_deps test fmt metalinter metalinter_all
+.PHONY: build install check_tools get_tools update_tools test
```

Comment:
add `lint`? why remove `fmt`?

I am against removing `get_vendor_deps`. We'll have to change all documentation + developers are forced to learn a new way! Why not just call `go modules install` or whatever instead of `dep ensure`?

Reply:
Hmm, the magic behind go modules is also that you do not need an extra step. `go build` or `go test` will fetch all the deps from mod.go / mod.sum for you (for go >= 1.11)... There is `go mod vendor` but  it isn't necessary. Also:
> By default, go commands like go build ignore the vendor directory when in module mode.
https://github.com/golang/go/wiki/Modules#how-do-i-use-vendoring-with-modules-is-vendoring-going-away

Maybe, we should update the documentation though?

Reply:
Oh.. I don't have to do anything, cool! Then we can drop `get_vendor_deps`, thanks 🙇

Reply:
> add lint? why remove fmt?

Added `lint` to `.PHONY`. I thought it might be better to run `fmt` through lint / golangci-lint. Added `fmt` to phony (as `gofmt` does a few more things than `go fmt` from Makefile).